src/Carbon/Translator.php

```diff
@@ -313,7 +313,7 @@ public function getMessages($locale = null)
      */
     public function setLocale($locale)
     {
-        $locale = preg_replace_callback('/[-_]([a-z]{2,})/', function ($matches) {
+        $locale = preg_replace_callback('/[-_]([a-z0-9]{2,})/', function ($matches) {
```

Comment:
I would like to be a bit more precise on this, e.g. only letter or only digits to be closer to the norm:
```suggestion
        $locale = preg_replace_callback('/[-_]([a-z]{2,}|[0-9]{2,})/', function ($matches) {
```

Reply:
This is actually how i started doing things then decided to go the other way `[a-z0-9]`   but i am happy with your suggestion